core/src/main/java/io/ap4k/utils/Generics.java

```diff
@@ -0,0 +1,77 @@
+package io.ap4k.utils;
+
+import java.lang.reflect.Array;
+import java.lang.reflect.GenericArrayType;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.lang.reflect.TypeVariable;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Generics {
+
+  public static Class getClass(Type type) {
+    if (type instanceof Class) {
+      return (Class)type;
+    } else if (type instanceof ParameterizedType) {
+      return getClass(((ParameterizedType)type).getRawType());
+    } else if (type instanceof GenericArrayType) {
+      Type componentType = ((GenericArrayType)type).getGenericComponentType();
+      Class<?> componentClass = getClass(componentType);
+      return componentClass != null ? Array.newInstance(componentClass, 0).getClass() : null;
+    } else {
+      return null;
+    }
+  }
+
+  public static <T> List<Class> getTypeArguments(Class baseClass, Class childClass) {
+    Map<Type, Type> resolvedTypes = new LinkedHashMap();
+    Object type = childClass;
+
+    while(true) {
+      int i;
+      while(!getClass((Type)type).equals(baseClass)) {
+        if (type instanceof Class) {
+          type = ((Class)type).getGenericSuperclass();
+        } else {
+          ParameterizedType parameterizedType = (ParameterizedType)type;
+          Class<?> rawType = (Class)parameterizedType.getRawType();
+          Type[] actualTypeArguments = parameterizedType.getActualTypeArguments();
+          TypeVariable<?>[] typeParameters = rawType.getTypeParameters();
+
+          for(i = 0; i < actualTypeArguments.length; ++i) {
+            resolvedTypes.put(typeParameters[i], actualTypeArguments[i]);
+          }
+
+          if (!rawType.equals(baseClass)) {
+            type = rawType.getGenericSuperclass();
+          }
+        }
+      }
+
+      Object actualTypeArguments;
+      if (type instanceof Class) {
+        actualTypeArguments = ((Class)type).getTypeParameters();
+      } else {
+        actualTypeArguments = ((ParameterizedType)type).getActualTypeArguments();
+      }
+
+      List<Class> typeArgumentsAsClasses = new ArrayList();
+      Object var12 = actualTypeArguments;
```

Comment:
I think we can drop `var12` :P

Reply:
We can't though I do need to copy the orginal code and not what intellij's decompiler makes out of the classes.

Reply:
Then let's call it something else :P. `var12 ` gives the generated feeling and I personally don't believe that's a good thing :)

Reply:
Already replaced with the original code.

Reply:
:+1: